pygdf/libgdf_groupby.py

```diff
@@ -0,0 +1,232 @@
+            
+#              assuming here that if there are multiple aggregations that the aggregated results will be in the same order for GDF_SORT method
+            if need_to_index:
+                out_col_indices_series = Series(Buffer(cuda.device_array(col_agg.size, dtype=np.int32)))
+                out_col_indices = out_col_indices_series._column.cffi_view
+            else:
+                out_col_indices = ffi.NULL
+
+            if first_run:
+                out_col_values_series = [Series(Buffer(cuda.device_array(col_agg.size, dtype=self._df[self._by[i]]._column.data.dtype))) for i in range(0,ncols)]
+                out_col_values = [out_col_values_series[i]._column.cffi_view for i in range(0,ncols)]
+            else :
+                out_col_values = ffi.NULL
+    
+            out_col_agg_series = Series(Buffer(cuda.device_array(col_agg.size, dtype=col_agg_dtype)))
+            out_col_agg = out_col_agg_series._column.cffi_view
+    
+            if agg_type == "min":
+                err = libgdf.gdf_group_by_min(
+                    ncols, cols, col_agg, out_col_indices, out_col_values, out_col_agg, ctx)
+            elif agg_type == "max":
+                err = libgdf.gdf_group_by_max(
+                    ncols, cols, col_agg, out_col_indices, out_col_values, out_col_agg, ctx)
+            elif agg_type == "count":
+                err = libgdf.gdf_group_by_count(
+                    ncols, cols, col_agg, out_col_indices, out_col_values, out_col_agg, ctx)
+            elif agg_type == "sum":
+                err = libgdf.gdf_group_by_sum(
+                    ncols, cols, col_agg, out_col_indices, out_col_values, out_col_agg, ctx)
+            elif agg_type == "mean":
+                err = libgdf.gdf_group_by_avg(
+                    ncols, cols, col_agg, out_col_indices, out_col_values, out_col_agg, ctx)
+            else:
+                print("ERROR: this aggregator has not been implemented yet")
+
+            if (err is not None):    
+                print(err)
+    
+            num_row_results = out_col_agg.size
+            
+            if first_run:
+                for i in range(0,ncols):
+                    out_col_values_series[i].data.size = num_row_results
+                    out_col_values_series[i] = out_col_values_series[i].reset_index()
+                    result[self._by[i]] = out_col_values_series[i]
+    
+            out_col_agg_series.data.size = num_row_results
+            out_col_agg_series = out_col_agg_series.reset_index()
+            
+            
+            if need_to_index:
+                out_col_indices_series.data.size = num_row_results
+                out_col_indices_series = out_col_indices_series.reset_index()
+                # TODO do something with the indices to align data
+            
+            
+            result[val_columns_out[col_count]] = out_col_agg_series  
+            
+            out_col_agg_series.data.size = num_row_results
```

Comment:
This line should be changed to `out_col_agg_series = out_col_agg_series[:num_row_results]`